presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueriesWithOptimizedRepartitioning.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.tests.AbstractTestDistributedQueries;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.sql.tree.ExplainType.Type.LOGICAL;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.tpch.TpchTable.getTables;
+import static org.testng.Assert.assertEquals;
+
+public class TestHiveDistributedQueriesWithOptimizedRepartitioning
```

Comment:
Do we really need to use `HiveQueryRunner`? -- A TPCH query runner should also do all the tests right? (Since it's based on `AbstractTestDistributedQueries`, and it's testing a core-engine feature so it doesn't relate which Connector you use :) )

Reply:
cc @arhimondr

Reply:
@wenleix The TPchQueryRunner doesn't support some features like these:

java.lang.RuntimeException: This connector does not support creating tables with data
java.lang.RuntimeException: This connector does not support creating views

Hundreds of queries fail with this query runner. We didn't want to overwrite all of the failed queries, so switched to HiveQueryRunner.

Reply:
@yingsu00 : I see. We should really enhance  `TpchQueryRunner`  to test such core engine features though, since now a lot of tests are done in `HiveQueryRunner` (namely, grouped execution...) and other connectors become second-class citizen 😞 .

But again this beyond the scope of this PR. So maybe just add a comment explains why we have to use `HiveQueryRunner` for this core-engine feature test.